cache/redis_cache.go

```diff
@@ -118,13 +119,17 @@ func (r *redisCache) Get(key *Key) (*CachedData, error) {
 		log.Errorf("Not able to fetch TTL for: %s ", key)
 	}
 
+	decoded, err := base64.StdEncoding.DecodeString(payload.Payload)
+	if err != nil {
+		log.Errorf("Not able to decode for: %s ", payload.Payload)
```

Comment:
Following this log, we should return from the function because payload is malformed. 
```suggestion
		log.Errorf("failed to decode payload: %s , due to: %v ", payload.Payload, err)
		return nil, ErrMissing
```

Reply:
thx 👍